### DISCUSSION TITLE
i need help python

### DISCUSSION THREAD

**Discussion by ze0m** [community/community#52440] [Programming Help]
### Body

i'm trying to create a macro program that allows me to type certain phrases at exactly that time, based on computer time, but it doesn't work well. what should i do?

import tkinter as tk
from datetime import datetime
import pyautogui
import time

class Unknown:
`    def __init__(self):`
        self.root = tk.Tk()
        self.root.title("unknown")
        self.root.geometry("250x100")
        
        self.label = tk.Label(self.root, text="")
        self.label.pack()
        
        self.button = tk.Button(self.root, text="button", command=self.update_time)
        self.button.pack(pady=10)
        
        self.update_time()
        self.root.mainloop()

    def update_time(self):
        now_micro = datetime.now().strftime("%H : %M : %S.%f")[:-4]
        if now_micro == "22 : 00 : 00.000":
            time.sleep(0.01)
            pyautogui.typewrite("complete")
        else:
            self.label.config(text=now_micro)
            self.root.after(10, self.update_time)
            
if __name__ == "__main__":
    Unknown()
    
### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by pie972** [comment]
Hello, use only root.after() and get rid of time.sleep() method in your tkinter application to schedule the typing at the specified time like this:
```
def update_time(self):
    now_micro = datetime.now().strftime("%H : %M : %S.%f")[:-4]
    if now_micro == "22 : 00 : 00.000":
        pyautogui.typewrite("complete")
    self.label.config(text=now_micro)
    self.root.after(10, self.update_time)
```
see if this solves the prob and let us know
